yt/go/bus/bus.go

```diff
@@ -347,12 +433,128 @@ func (c *Bus) receive(message io.Reader) (busMsg, error) {
 	}, nil
 }
 
+func (c *Bus) receiveHandshake() (EncryptionMode, error) {
+	e := EncryptionModeDisabled
+
+	msg, err := c.receive()
+	if err != nil {
+		return e, fmt.Errorf("bus: error receiving handshake: %w", err)
+	}
+
+	if msg.fixHeader.typ != packetMessage {
+		return e, fmt.Errorf("bus: handshake type mismatch")
+	}
+
+	if msg.fixHeader.partCount != 1 {
+		return e, fmt.Errorf("bus: handshake part count mismatch")
+	}
+
+	if msg.fixHeader.packetID != guid.FromParts(1, 0, 0, 0) {
+		return e, fmt.Errorf("bus: handshake packet id mismatch")
+	}
+
+	data := msg.parts[0]
+	if len(data) < 4 {
+		return e, fmt.Errorf("bus: handshake data too small")
+	}
+
+	signature := binary.LittleEndian.Uint32(data[0:4])
+	if signature != handshakeSignature {
+		return e, fmt.Errorf("bus: handshake data signature mismatch")
+	}
+
+	var handshake THandshake
+	if err := proto.Unmarshal(data[4:], &handshake); err != nil {
+		return e, fmt.Errorf("bus: handshake data unmarshal error: %w", err)
+	}
+
+	e = EncryptionMode(handshake.GetEncryptionMode())
+	c.logger.Debug("Handshake received",
+		log.String("encryptionMode", fmt.Sprintf("%d", e)))
```

Comment:
In other clients we usually use snake case for log fields: `encryptionMode -> enciption_mode`.